src/components/settings/SettingsLMSTab/LMSConfigs/utils.tsx

```diff
@@ -175,13 +175,18 @@ export async function handleSaveHelper(
   return !err;
 }
 
-export function checkForDuplicateNames(existingConfigNames: string[]): FormFieldValidation {
+export function checkForDuplicateNames(existingConfigNames: Map<string, string>): FormFieldValidation {
     return {
     formFieldId: 'displayName',
     validator: (fields) => {
-      return existingConfigNames?.includes(fields['displayName'])
-        ? INVALID_NAME
-        : false;
+      let validName = true;
+      validName = !(existingConfigNames?.has(fields['displayName']));
+      if (fields.id && !validName) { // if we're editing an existing config
+        if (existingConfigNames.get(fields['displayName']) == fields.id) {
+          validName = true;
+        }
+      }
+      return validName ? '' : INVALID_NAME;
```

Comment:
Should use 'false' for validation that passes.  Technically, a falsy string does work at the moment ([since we're just filtering by truthiness without type checking](https://github.com/openedx/frontend-app-admin-portal/blob/013d73b106b22fd05e3295ac814b500f54517e0b/src/components/forms/data/reducer.ts#L27-L32)), but that might change in the future.
```suggestion
      return validName ? false : INVALID_NAME;
```

Reply:
I just did this because typescript throws warnings when it expects a string but gets a boolean. Should I just ignore that error?

Reply:
What's the error?  The return value of the Function explicitly accepts boolean.
https://github.com/openedx/frontend-app-admin-portal/blob/0a98549f38f5f1e42bb9c496a3d6b3f2c3b6b09e/src/components/forms/FormContext.tsx#L6-L11